src/trio/_core/_tests/test_io.py

```diff
@@ -381,6 +384,38 @@ def check(*, expected_readers: int, expected_writers: int) -> None:
         check(expected_readers=1, expected_writers=0)
 
 
+@pytest.mark.skipif(sys.platform in {"win32", "linux"}, reason="no kqueue")
```

Comment:
From mypy run annotations, looks like you need to inform mypy that you don't expect this function to run on windows or linux, because mypy doesn't understand `pytest.mark.skipif`.
I suggest something like this:
```python
if sys.platform == "win32" or sys.platform == "linux":
    raise AssertionError("Should be unreachable")
```

Reply:
@CoolCat467 perhaps, `assert_never()`?

Reply:
Or better yet `assert sys.platform not in {"win32", "linux"}`

Reply:
I couldn't get either option working (assert sys.platform != "win32" and sys.platform != "linux") does work in the playground, and if I create a new file.